corehq/apps/accounting/models.py

```diff
@@ -500,13 +507,24 @@ def _get_account_by_created_by_domain(cls, domain):
     def get_enterprise_restricted_signup_accounts(cls):
         return BillingAccount.objects.filter(is_customer_billing_account=True, restrict_signup=True)
 
+    @classmethod
+    def get_enterprise_permissions_domains(cls, domain):
```

Comment:
considering that this is called in places like middleware I think it should be cached.

Reply:
7ed5564a280d32f0eb3bdee2d2e4c3503f5a5fb4